### ISSUE TITLE
Singularities in the euler-angle floating base joint cause infrequent errors on unit tests with random initial conditions

### ISSUE THREAD

**Issue by RussTedrake** [RobotLocomotion/drake#18]
The solution, of course, is to implement the floating base with a non-degenerate coordinate system (e.g. quaternions).  But this is painful for a number of reasons:  
* it's not a minimal coordinate system, so all of the dependent analysis and feedback design methods have to reason about the extra constraint.
* it means that rigid body systems are not second order systems anymore.  (and num_states ~= 2*num_dof).  unfortunately this assumption is baked into a lot of places, so the change will have be applied carefully.

```Matlab
<matlab_test> (strcmp(error_id,'PathLCP:FailedToSolve') || ...
    (strcmp(error_id,'Simulink:SFunctions:SFcnErrorStatus') && (~isempty(strfind(test_output,'Path fails')) || ~isempty(strfind(test_output,'not finite')))) ...
  && any(strcmp(test_name,{'systems/plants/test/fallingBrickLCP', ...
          'systems/plants/test/fallingCapsulesTest',...
          'systems/plants/test/multiRobotTest', ...
          'systems/plants/test/momentumTest', ...
          'systems/plants/test/terrainTest', ...
          'systems/plants/test/terrainInterpTest', ...
          'systems/plants/test/testFloatingBaseDynamics', ...
          'examples/PR2/runPassive',...
          'examples/Quadrotor/Quadrotor.runOpenLoop', ...
          'examples/Quadrotor/test/testCollisions', ...
          'examples/ZMP/CartTable.run'})))</matlab_test>
```

**Comment by RussTedrake** [comment]
blocked by #1013, which is not a priority to resolve until after the DRC

**Comment by RussTedrake** [comment]
resolved by #1375

**Comment by avalenzu** [comment]
Wow! The end of an era ;)

**Comment by tkoolen** [comment]
Yeah, pretty awesome!
